src/lib.rs

```diff
@@ -29,27 +29,62 @@ pub use crate::node::Taffy;
 #[cfg(feature = "std")]
 use core::fmt::{Display, Formatter, Result};
 
-/// An error that can occur when performing layout
+/// The [`Node`](node::Node) was not found in the [`Taffy`] instance
 #[derive(Debug)]
-pub enum Error {
-    /// The [`Node`](node::Node) was invalid
-    InvalidNode(node::Node),
+pub struct NodeNotFoundError(pub node::Node);
+
+#[cfg(feature = "std")]
+impl Display for NodeNotFoundError {
+    fn fmt(&self, f: &mut Formatter) -> Result {
+        write!(f, "Node {:?} is not in the Taffy instance", self.0)
+    }
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for NodeNotFoundError {}
+
+/// An error that occurs while trying to access or modify a [`Node`](node::Node)'s children by index.
+#[derive(Debug)]
+pub enum ChildOperationError {
+    /// The parent [`Node`](node::Node) does not have a child at `child_index`. It only has `child_count` children
+    ChildIndexOutOfBounds {
+        /// The parent node whose child was being looked up
+        parent: node::Node,
+        /// The index that was looked up
+        child_index: usize,
+        /// The total number of children the parent has
+        child_count: usize,
+    },
+    ///The [`Node`](node::Node) was not found in the [`Taffy`] instance
+    NodeNotFoundError(NodeNotFoundError),
```

Comment:
IMO we can just keep this as an enum variant; no need for a seperate type.
```suggestion
    NodeNotFound(node::Node),
```